Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -0,0 +1,1101 @@
+            {
+                AZ::Data::AssetId matchingPhysicsAsset = FindMatchingPhysicsAsset(renderMeshAsset, physicsAssets);
+
+                if (matchingPhysicsAsset.IsValid())
+                {
+                    SetMeshAsset(matchingPhysicsAsset);
+                }
+                else
+                {
+                    AZ_Warning("EditorMeshColliderComponent", false,
+                        "SetCollisionMeshFromRender on entity %s: Unable to find a matching physics asset "
+                        "for the render mesh asset GUID: %s, hint: %s",
+                        GetEntity()->GetName().c_str(),
+                        renderMeshAsset.GetId().m_guid.ToString<AZStd::string>().c_str(),
+                        renderMeshAsset.GetHint().c_str());
+                }
+            }
+            // This is not necessarily an incorrect case but it's worth reporting 
+            // in case if we forgot to configure the source asset to produce the collision mesh
+            else if (physicsAssets.empty())
+            {
+                AZ_TracePrintf("EditorMeshColliderComponent",
+                    "SetCollisionMeshFromRender on entity %s: The source asset for %s did not produce any physics assets",
+                    GetEntity()->GetName().c_str(),
+                    renderMeshAsset.GetHint().c_str());
+            }
+        }
+        else
+        {
+            AZ_Warning("EditorMeshColliderComponent", false, 
+                "SetCollisionMeshFromRender on entity %s: Unable to get the assets produced by the render mesh asset GUID: %s, hint: %s",
+                GetEntity()->GetName().c_str(), 
+                renderMeshAsset.GetId().m_guid.ToString<AZStd::string>().c_str(),
+                renderMeshAsset.GetHint().c_str());
+        }
+    }
+
+    void EditorMeshColliderComponent::OnModelReady(
+        [[maybe_unused]] const AZ::Data::Asset<AZ::RPI::ModelAsset>& modelAsset,
+        [[maybe_unused]] const AZ::Data::Instance<AZ::RPI::Model>& model)
+    {
+        if (ShouldUpdateCollisionMeshFromRender())
+        {
+            SetCollisionMeshFromRender();
+        }
+    }
+
+    void EditorMeshColliderComponent::SetAssetScale(const AZ::Vector3& scale)
+    {
+        m_shapeConfiguration.m_configuration.m_assetScale = scale;
+        UpdateCollider();
+    }
+
+    AZ::Vector3 EditorMeshColliderComponent::GetAssetScale() const
+    {
+        return m_shapeConfiguration.m_configuration.m_assetScale;
+    }
+
+    void EditorMeshColliderComponent::UpdateShapeConfiguration()
+    {
+        UpdateShapeConfigurationScale();
+    }
+
+    AZ::Aabb EditorMeshColliderComponent::GetWorldBounds()
+    {
+        return GetAabb();
+    }
+
+    AZ::Aabb EditorMeshColliderComponent::GetLocalBounds()
+    {
+        AZ::Aabb worldBounds = GetWorldBounds();
+        if (worldBounds.IsValid())
+        {
+            return worldBounds.GetTransformedAabb(m_cachedWorldTransform.GetInverse());
+        }
```

Comment:
comment: I realize this was probably like this before but it does seem odd we get the local bounds from the world bounds instead of the other way around 😅 not a huge deal though

Reply:
PhysX only provides the bounds in world space, you have to do it this way round unless you want some very complex (and probably fairly expensive) logic for computing the local bounds.

Reply:
I see got you thanks! 👍